Add tests

martins-vds opened the pull request:
This PR adds tests and makes small refactors

stebet commented on Functions/Hash.cs:
```diff
@@ -18,6 +18,11 @@ public static class Hash
         /// <returns>The input string as a SHA-1 hash</returns>
         public static string CreateSHA1Hash(this string input, string source = "UTF8")
         {
+            if (string.IsNullOrWhiteSpace(input))
```
This should be redundant with nullable checks enabled. This method should never get a null as input. Also, whitespace is a legitimate string to hash :)

stebet commented on Functions/Hash.cs:
```diff
@@ -33,14 +38,21 @@ public static string CreateSHA1Hash(this string input, string source = "UTF8")
         public static bool IsHexStringOfLength(this string input, int requiredLength)
         {
-            if (input.Length == 0 || input.Length != requiredLength)
+            if (string.IsNullOrWhiteSpace(input))
```
Whitespace check is unnecessary since it'll fail the HexChar test. This method will also never receive null as input.

stebet commented on Functions/Hash.cs:
```diff
@@ -33,14 +38,21 @@ public static string CreateSHA1Hash(this string input, string source = "UTF8")
+                return false;
+            }
+
+            var sanitizedInput = input.Trim();
```
No need to trim. Whitespace fails the IsHex check anyway.

stebet reviewed (changes requested):
You can remove the unneeded null and whitespace checks

stebet commented:
Love getting tests for the function itself!

martins-vds commented:
> You can remove the unneeded null and whitespace checks

Thanks for reviewing the code, @stebet !

stebet commented on Functions.Tests/Helpers/TestFunctionContext.cs:
```diff
@@ -0,0 +1,65 @@
+            FunctionDefinition = functionDefinition;
+            _invocation = invocation;
+
+            //Features.Set<IFunctionBindingsFeature>(new TestFunctionBindingsFeature
+            //{
+            //    OutputBindingsInfo = new DefaultOutputBindingsInfoProvider().GetBindingsInfo(FunctionDefinition)
+            //});
```
Remove if unused.

stebet commented on Functions.Tests/Helpers/TestFunctionDefinition.cs:
```diff
@@ -0,0 +1,85 @@
+
+            for (int i = 0; i < paramTypes.Length; i++)
+            {
+                var properties = new Dictionary<string, object>
+                {
+                    {"TestPropertyKey", "TestPropertyValue" }
+                };
```
Is there a point to this dictionary?

mdawsonuk replied:
Looks like it's used in line 75

stebet replied:
Yeah but does it need to be populated? Can it be static?

stebet commented:
@martins-vds Can you merge the latest changes from `main` so we can get this merged :)

stebet commented on Functions.Tests/Helpers/TestFunctionContext.cs:
```diff
@@ -0,0 +1,65 @@
+            FunctionDefinition = functionDefinition;
+            _invocation = invocation;
+
+            //Features.Set<IFunctionBindingsFeature>(new TestFunctionBindingsFeature
+            //{
+            //    OutputBindingsInfo = new DefaultOutputBindingsInfoProvider().GetBindingsInfo(FunctionDefinition)
+            //});
```
```suggestion
```

martins-vds replied:
Code removed

martins-vds commented:
> @martins-vds Can you merge the latest changes from `main` so we can get this merged :)

All done! 😃

stebet commented:
> > @martins-vds Can you merge the latest changes from `main` so we can get this merged :)
> 
> 
> 
> All done! 😃 

Something is off with the revisions. It's marking files as changed, such as the workflow files. Is your fork out-of-sync?

martins-vds commented:
> > > @martins-vds Can you merge the latest changes from `main` so we can get this merged :)
> > 
> > 
> > All done! 😃
> 
> Something is off with the revisions. It's marking files as changed, such as the workflow files. Is your fork out-of-sync?

That's weird. I updated my fork this morning before I made any change and rebased this branch onto main

martins-vds commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,8 @@
+version: 2
```
Nope, just a regular push

martins-vds replied:
I'll try and recreate this branch to see if those files go away

stebet replied:
> Nope, just a regular push

That might be the issue then :) If you have already pushed changes to your remote before the rebase, you usually need to force-push after rebasing to override the older pre-rebase changes.

martins-vds commented:
Closing this PR in favor of #42
